### PR TITLE
ESEF 2023 Conformance Suite

### LINKED ISSUE #893: ESEF 2023 Validation Updates
### What should we change and why?

The [ESEF reporting manual](https://www.esma.europa.eu/sites/default/files/library/esma32-60-254_esef_reporting_manual.pdf) has been updated for 2023. The following table tracks rule changes and potential updates for the ESEF Arelle plugin.

| Rule Number | Rule Name                                                                                              | Change information                                                                                                                                                                                                                                                                                       | Workiva Comments                                                                                                                                                                                                                                                               | Acsone Comments                                                                                                                                                                                                                           |
| ------------------------------------------------------------------------------------------------------------- | ------------------------------------------------------------------------------------------------------ | -------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | -------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
| Guidance 1.2.2                                                                                                | Use of elements available in the IFRS Taxonomy that were not yet included in the ESEF taxonomy         | Added clarification and example about how to extend for elements that exist in future taxonomies                                                                                                                                                                                                         |                                                                                                                                                                                                                                                                                  |                                                                                                                                                                                                                                           |
| Guidance 1.4.1                                                                                                | Anchoring of extension elements to elements in the ESEF taxonomy that are wider in scope or meaning    | Call out on when to not create extensions                                                                                                                                                                                                                                                                |                                                                                                                                                                                                                                                                                  |                                                                                                                                                                                                                                           |
| Guidance 1.9.1                                                                                                | Marking up notes and accounting policies                                                               | Removing the use of the 'broad' text block tag (that element is removed in 2023 IFRS taxonomy) on all notes<br><br>Added note that filers can use elements from Annex VI to compliment tagging                                                                                                           | Could be consideration item as a warning to no longer use 'DisclosureOfNotesAndOtherExplanatoryInformationExplanatory' in a filing, but would be temporary until the next taxonomy version comes out. May not be significant impact for data usage and not critical to implement | Yes, there could be deprecation warning if 'DisclosureOfNotesAndOtherExplanatoryInformationExplanatory' is still used.<br>We shall make this ckeck optional, meaning that it can be disabled if some national authority does not want it. |
| Guidance 1.9.2                                                                                                | Granularity of block tagging of notes and accounting policies                                          | Removing the use of the 'broad' text block tag (that element is removed in 2023 IFRS taxonomy) on all notes<br><br>Added note about ensuring following guidance of text block tagging and the images are illustrations                                                                                   |                                                                                                                                                                                                                                                                                  |                                                                                                                                                                                                                                           |
| Guidance 1.9.3                                                                                                | Other considerations for block tagging of notes and accounting policies                                | Removing the use of the 'broad' text block tag (that element is removed in 2023 IFRS taxonomy) on all notes<br><br>Added that issuers can detail tag the notes if they want.                                                                                                                             |                                                                                                                                                                                                                                                                                  |                                                                                                                                                                                                                                           |
| Guidance 2.2.5                                                                                                | Tagging of dashes or empty fields                                                                      | Updated wording about tagging blank cells or dash values when representing financial info like zero. Noted that values that represent nil will need to be explicitly set as there is no transformation for this                                                                                          |                                                                                                                                                                                                                                                                                  |                                                                                                                                                                                                                                           |
| Guidance 2.2.6                                                                                                | Readability of the information extracted from Technical construction of a block tag                    | Changed the name of the rule<br><br>Updated wording for more call out about xhtml rendered out side the doc sould be legible and clear, but doesn't have to resemble style. Information that is contained in tables in the doc should be<br>meaningfully transcribed in the extracted tagged information |                                                                                                                                                                                                                                                                                  |                                                                                                                                                                                                                                           |
| Guidance 2.2.7                                                                                                | Technical construction of a block tag                                                                  | NEW: This split some of the info from prior 2.2.6 and moves the related validation here. This includes the guidance to better resemble the extracted tag info.                                                                                                                                           | Looks like suggested validation "Violation: escapedHTMLUsedInBlockTagWithSpecialCharacters" was moved from 2.2.6 to 2.2.7. Reference may need to be changed                                                                                                                      | Yes, we confirm that the reference needs to be changed                                                                                                                                                                                    |
| Guidance 2.5.1                                                                                                | Inclusion of content other than XHTML and XBRL in the Inline XBRL document                             | Adds note - 'please note that direct embedding of <svg> elements is not allowed and the SVG images shall be included in <img> element'<br><br>Made wording change for RfC to RFC<br>Removed JPG from image options                                                                                       | JPG was removed from the format options related to suggested validation for “imageFormatNotSupported”<br><br>With added note that direct embedding of <svg> files is not allowed and the SVG images shall be included in <img> element - may need validation update              | Error message for 'imageFormatNotSupported' does not mention 'JPG'. No update is needed.<br>Direct SVG is already not allowed in XHTML documents.                                                                                         |
| Guidance 2.6.1                                                                                                | Including Inline XBRL document in taxonomy packages                                                    | Updated to the 2023 report package specification from Working Group Note                                                                                                                                                                                                                                 | Recommended Violation: “reportIncorrectlyPlacedInPackage” may need to reference updated spec link                                                                                                                                                                                | Yes, we confirm that the reference needs to be changed                                                                                                                                                                                    |
| Guidance 2.6.2                                                                                                | Including multi-html Inline XBRL documents and multiple Inline XBRL document sets in taxonomy packages | Updated to the 2023 report package specification from Working Group Note                                                                                                                                                                                                                                 |                                                                                                                                                                                                                                                                                  | Same as above                                                                                                                                                                                                                             |
| Guidance 3.1.3                                                                                                | Taxonomy packages                                                                                      | Minor wording change to remove 'Working Group Note' and now only specification wording                                                                                                                                                                                                                   |                                                                                                                                                                                                                                                                                  |                                                                                                                                                                                                                                           |
| Guidance 3.2.2                                                                                                | Data types to be used on extension concepts                                                            | Updated domainItemType defined in DTR to https://www.xbrl.org/dtr/type/2020-01-21/types.xsd                                                                                                                                                                                                              | Recommended Violation: “domainMemberWrongDataType” may need to be updated to reference new DTR                                                                                                                                                                                   | Yes, we confirm that the reference needs to be changed                                                                                                                                                                                    |
| Guidance 3.3.1                                                                                                | Relationships to anchor extension taxonomy elements to elements in the ESEF taxonomy                   | Minor wording change switching order of sentence                                                                                                                                                                                                                                                         |                                                                                                                                                                                                                                                                                  | Important to avoid wider-narrower warning for abstract elements.<br>Code will be changed to only issue a warning if the new element is not abstract.                                                                                      |
| Guidance 3.4.1                                                                                                | Modelling of the issuers’ extension taxonomies’ linkbases                                              | Added language about the futue of Calc 2.0 and status of Calc 1.1. At this time, they discourage use of Calc 1.1, but will continue to monitor                                                                                                                                                           |                                                                                                                                                                                                                                                                                  | It is important to continue using Calc 1.0.<br>We have to add a new check to warn if Calc 1.1 roles are used, their usage is discouraged.                                                                                                 |
| Guidance 3.4.2                                                                                                | Defining the dimensional validity of line items in the definition linkbase                             | Updating url of hypercube for line items with no dimension. Added "s" for https                                                                                                                                                                                                                          | This fixed a missing "s" from the url. Not sure impact to existing validations or what it references. Would be now: https://www.esma.europa.eu/xbrl/role/cor/esef_role-999999                                                                                                    | Yes, there are hardcoded references in the source code; an update of the source code will be needed.                                                                                                                                      |
| Guidance 3.4.3                                                                                                | Definition of default members of extension taxonomy dimensions                                         | Updating url of hypercube for default member assignment. Added "s" for https                                                                                                                                                                                                                             | This fixed a missing "s" from the url. Not sure impact to existing validations or what it references. Would be now: https://www.esma.europa.eu/xbrl/role/cor/ifrsdim_role-990000                                                                                                 | Same as above                                                                                                                                                                                                                             |
| Guidance 4.1.1                                                                                                | Reporting of stand-alone of xHTML files                                                                | Updated langugae to what 'multiple files' represents - "a single XHTML file and any associated referenced images"                                                                                                                                                                                        |                                                                                                                                                                                                                                                                                  | We suggest adding a test to explicitly disallow multiple XHTML files in stand-alone ZIP archives.                                                                                                                                         |
| Guidance 4.1.3                                                                                                | Inclusion of content other than XHTML in a stand-alone XHTML file                                      | Removed validation note for 'imageIncludedAndNotEmbeddedAsBase64EncodedString”<br><br>Also adds that RFC 2045 is to be used together with RFC 2046 and RFC 2048.<br><br>Adds the same language as 2.5.1 about svg                                                                                        | Removed Recommended Violation: “imageIncludedAndNotEmbeddedAsBase64EncodedString”<br><br>Similar to to 2.5.1, removed JPG from format options and added note about direct embedding <svg> is not allowed. Not sure if validations are duplicated within their set up.<br>        | Yes, we confirm that error message must be removed.<br>The direct embedding of SVG elements was never allowed. Nothing to do for that change.                                                                                             |

### PR DESCRIPTION
**Pull request by austinmatherne-wk**
#### Reason for change
ESEF 2023 conformance suite was released.
Resolves #893 

#### Description of change
* Resolve compatibility issues with ESEF 2023 conformance suite.
* Add ESEF 2023 conformance suites to CI.
* Mark broken test cases as expected failures.
* Mark ESEF 2023 disclosure systems as default ESEF disclosure systems.

#### Steps to Test
* CI

**review**:
@Arelle/arelle
@gmongelli

### CODE REVIEW

**Comment by aviary2-wf** [comment]
## Security Insights
No security relevant content was detected by automated scans.

## Action Items
* Review PR for [security impact](https://wiki.atl.workiva.net/display/SECURITY/Development+Security+Review+Guidelines); comment "security review required" if needed or unsure
* Verify [`aviary.yaml`](https://wiki.atl.workiva.net/pages/viewpage.action?pageId=42762752#SecurityInsightFramework&Tooling-ConfiguringAviary) coverage of security relevant code
---
_Questions or Comments? Reach out on Slack: #support-infosec._

File: arelle/plugin/validate/ESEF/ESEF_Current/ValidateXbrlFinally.py [hunk 1]
```diff
@@ -147,19 +147,6 @@ def validateXbrlFinally(val: ValidateXbrl, *args: Any, **kwargs: Any) -> None:
...
-        instanceNumber = 0
-        if modelXbrl.fileSource.dir:
-            for file in modelXbrl.fileSource.dir:
-                if not file.endswith("/"):
-                    from arelle.ModelDocument import Type
-                    fileType = Type.identify(modelXbrl.fileSource, "{}/{}".format(modelXbrl.fileSource.basefile, file))
-                    if fileType == Type.INLINEXBRL:
-                        instanceNumber += 1
-            if instanceNumber > 1:
-                modelXbrl.error("ESEF.4.1.1.SingleXhtmlFiles",
```
Comment (austinmatherne-wk) [inline arelle/plugin/validate/ESEF/ESEF_Current/ValidateXbrlFinally.py:159]: Test cases for `2.5.4` and `2.6.2` both explicitly check multi xhtml report packages.

File: arelle/plugin/validate/ESEF/__init__.py [hunk 2]
```diff
@@ -121,21 +117,27 @@ def modelDocumentPullLoader(
...
-                        ModelDocument.Type.identify(modelXbrl.fileSource, cast(str, modelXbrl.fileSource.url)) in (
-                            ModelDocument.Type.TESTCASESINDEX, ModelDocument.Type.TESTCASE)):
+                if isinstance(modelXbrl.fileSource.url, str) and modelXbrl.fileSource.url.endswith(".xml"):
+                    documentType = ModelDocument.Type.identify(modelXbrl.fileSource, modelXbrl.fileSource.url)
+                    if documentType in {ModelDocument.Type.TESTCASESINDEX, ModelDocument.Type.TESTCASE}:
                         return None  # allow zipped test case to load normally
+
+                if disclosureSystemYear >= 2023 and not modelXbrl.fileSource.isZip:
+                    modelXbrl.error("ESEF.2.6.1.disallowedReportPackageFileExtension",
+                                    _("A report package MUST conform to the .ZIP File Format Specification and MUST have a .zip extension."),
```
Comment (brettkail-wk) [inline arelle/plugin/validate/ESEF/__init__.py:136]: Out of curiosity, where does the "MUST have a .zip extension" come from?

**Review by gmongelli — CHANGES_REQUESTED** [review]

**Review by brettkail-wk — APPROVED** [review]

File: arelle/plugin/validate/ESEF/__init__.py (hunk 2 above)
Comment (brettkail-wk) [reply arelle/plugin/validate/ESEF/__init__.py:136]: Thanks.  Interesting that it's not mentioned by the filing manual...

**Review by derekgengenbacher-wf — DISMISSED** [review]

File: arelle/plugin/validate/ESEF/__init__.py (hunk 2 above)
Comment (gmongelli) [reply arelle/plugin/validate/ESEF/__init__.py:136]: Well, in the ESEF reporting manual, the only reference I see is this one (2.6.1):

> ESMA recommends that issuers follow the recommendations of XBRL International,  which indicates how Inline XBRL documents are to be included within a taxonomy package.

The specifications of a taxonomy package extension are at

Comment (austinmatherne-wk) [reply arelle/plugin/validate/ESEF/__init__.py:136]: 2.6.3 speaks to it a bit, but doesn't describe it as a requirement like the conformance suite does:
> Whilst ESMA did not define in the RTS on ESEF a unique naming convention for ESEF
files, unless the relevant Officially Appointed Mechanism and / or National Competent
Authorities provide indications of any specific naming conventions which are required
at national level, ESMA encourages issuers to adopt a naming convention which match
{base}-{date}-{lang}.zip, whereby:
> ...
> The above naming convention is recommended both for the report package files (with .zip extension) and for any report file (with .html or .xhtml extension) present within the package. For the naming convention of the taxonomy files that are part of the report package, please refer to Guidance 3.1.5.

**Review by derekgengenbacher-wf — APPROVED** [review]

**Pull request merged**